### DISCUSSION TITLE
Simulator not working on Linux

### DISCUSSION THREAD

**Discussion by RustyStriker** [zepp-health/zeppos-forum#103] [Q&A]
I have installed the simulator, and after verifying i have all the needed packages(and updating the `qemu-system-arm` binary to the latest one, in order to run it on my system), but it appears that when i try to start an application(or just open the vm itself) it requests my root password in the terminal(and other times it claims it cant find the gtk module although installed)

**Comment by MyoungXUE** [comment]
Thank you for your feedback, I will feedback this issue to the relevant colleagues.

**Comment by jiaolaibin** [comment]
Yes, it requires a root password because the emulator uses 'tun/tap' to access the local network.

The command "**ldd qemu-system-arm**" can check if the gtk library is missing, and the graphical interface display library can also use 'sdl2' as an alternative;

Please tell me your linux publisher and version number.

**Reply by RustyStriker** [reply]
im using Artix linux with kernel version `6.2.2-artix-1`, the `ldd` command shows everything is in place.

It seems a bit concerning that the simulator will require my root password, is there a way to circumvent it?

**Reply by jiaolaibin** [reply]
The emulator needs the root password just to create a virtual network interface card, nothing else.

**Reply by RustyStriker** [reply]
managed to make qemu "work" by using the system's qemu binary directly, which faults at startup with `qemu: fatal: Lockup: can't escalate 3 to HardFault` as the error message(plus a small dump), it seems i cannot build the qemu in the zepp repositories.. can i just use my watch directly as a testing ground at this point?

**Reply by jiaolaibin** [reply]
using the system's qemu binary？Can you tell me how to use it?

**Reply by jiaolaibin** [reply]
Don't try to use the qemu downloaded from the official website, the qemu in our emulator has added a lot of customized modifications.
You can find it in the directory "/opt/simulator/resources/firmware/".

**Reply by RustyStriker** [reply]
> using the system's qemu binary？Can you tell me how to use it?

well, i just replaced the qemu string to the one in my system(in `/usr/bin`), but it means its using the "official" qemu. i am trying to compile the qemu from your repos, but it doesnt seem to work properly, can you write an update build instructions ?

**Reply by jiaolaibin** [reply]
I suggest you use qemu directly under the simulator, you can find it in the directory "/opt/simulator/resources/firmware/".

If you need to build, download our repository(https://github.com/zepp-health/qemu) and try the following steps:
**cd qemu/qemu
mkdir -p bin/release
cd bin/release
../../configure --target-list=arm-softmmu --disable-debug-info --disable-qom-cast-debug
make**

You can also refer to the documentation: https://wiki.qemu.org/Hosts/Linux.
